File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    static final String YES = "Yes";
    static final String NO = "No";
    
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int pointer1 = 0;
        int pointer2 = 0;
        
        for (String word : goal) {
            if (cards1[pointer1].equals(word)) {
                pointer1++;
            } else if (cards2[pointer2].equals(word)) {
                pointer2++;
            } else {
                return NO;
            }
        }
        
        return YES;
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.kt
class Solution {
    companion object {
        private const val YES = "Yes"
        private const val NO = "No"
    }
    
    fun solution(cards1: Array<String>, cards2: Array<String>, goal: Array<String>): String {
        var pointer1 = 0
        var pointer2 = 0
        
        for (word in goal) {
            if (pointer1 < cards1.size && cards1[pointer1] == word) {
                pointer1++
            } else if (pointer2 < cards2.size && cards2[pointer2] == word) {
                pointer2++
            } else {
                return NO
            }
        }
        
        return YES
    }
}